sql/analyzer/process.go

```diff
@@ -0,0 +1,174 @@
+package analyzer
+
+import (
+	"fmt"
+	"hash/crc32"
+	"io"
+
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+	"gopkg.in/src-d/go-mysql-server.v0/sql/plan"
+)
+
+func trackProcess(ctx *sql.Context, a *Analyzer, n sql.Node) (sql.Node, error) {
+	if !n.Resolved() {
+		return n, nil
+	}
+
+	if _, ok := n.(*queryProcess); ok {
+		return n, nil
+	}
+
+	var query string
+	if x, ok := ctx.Value(sql.QueryKey).(string); ok {
+		query = x
+	}
+
+	var typ = sql.QueryProcess
+	if _, ok := n.(*plan.CreateIndex); ok {
+		typ = sql.CreateIndexProcess
+	}
+
+	processList := a.Catalog.ProcessList
+	pid := processList.AddProcess(ctx, typ, query)
+
+	var seen = make(map[string]struct{})
+	wrapTable := func(name string, t sql.Table) (sql.Table, error) {
+		if _, ok := seen[name]; ok {
+			return t, nil
+		}
+
+		var total int64 = -1
+		if counter, ok := t.(sql.PartitionCounter); ok {
+			count, err := counter.PartitionCount(ctx)
+			if err != nil {
+				return nil, err
+			}
+			total = count
+		}
+		processList.AddProgressItem(pid, name, total)
+
+		seen[name] = struct{}{}
+		return &processTable{t, func() {
+			processList.UpdateProgress(pid, name, 1)
+		}}, nil
+	}
+
+	n, err := n.TransformUp(func(n sql.Node) (sql.Node, error) {
+		switch n := n.(type) {
+		case *plan.ResolvedTable:
+			if _, ok := n.Table.(*processTable); ok {
+				return n, nil
+			}
+
+			t, err := wrapTable(n.Name(), n.Table)
+			if err != nil {
+				return nil, err
+			}
+
+			return plan.NewResolvedTable(n.Name(), t), nil
+		case sql.Table:
```

Comment:
is this necessary? Why we cannot just get ResolvedTables?

Reply:
Not all tables may be inside ResolvedTables (e.g. SquashedTables)

Reply:
So, if I understood correctly, maybe the easiest solution is to move the table name to the sql.Table interface, and apply the rule just to `sql.Table`s, right?

Reply:
Yup. Not having the name in the Table is a PITA a lot of times, and there's no reason a table can't know its name (after all, it needs to return it as the `Source` in the `Schema` method, so it must know).